SU2_CFD/src/solvers/CTurbSASolver.cpp

```diff
@@ -250,6 +250,52 @@ void CTurbSASolver::Postprocessing(CGeometry *geometry, CSolver **solver_contain
   }
   END_SU2_OMP_FOR
 
+
+  /*--- Compute turbulence index ---*/
+  if(config->GetKind_Trans_Model() != TURB_TRANS_MODEL::NONE || config->GetSAParsedOptions().bc) {
+    auto* flowNodes = su2staticcast_p<CFlowVariable*>(solver_container[FLOW_SOL]->GetNodes());
+
+    for (auto iMarker = 0; iMarker < config->GetnMarker_All(); iMarker++)
+      switch (config->GetMarker_All_KindBC(iMarker)) {
+        case ISOTHERMAL:
+        case HEAT_FLUX:
+        case HEAT_TRANSFER:
+          SU2_OMP_FOR_STAT(OMP_MIN_SIZE)
+          for (auto iVertex = 0u; iVertex < geometry->nVertex[iMarker]; iVertex++) {
+            const auto iPoint = geometry->vertex[iMarker][iVertex]->GetNode();
+
+            /*--- Check if the node belongs to the domain (i.e, not a halo node) ---*/
+
+            if (geometry->nodes->GetDomain(iPoint)) {
+              const auto jPoint = geometry->vertex[iMarker][iVertex]->GetNormal_Neighbor();
+
+              su2double FrictionVelocity = 0.0;
+              if(nDim == 2){
+                su2double shearStress = 0.0;
+                for(auto iDim = 0u; iDim < nDim; iDim++) {
+                  shearStress += pow(solver_container[FLOW_SOL]->GetCSkinFriction(iMarker, iVertex, iDim), 2.0);
+                }
+                shearStress = sqrt(shearStress);
+
+                FrictionVelocity = sqrt(shearStress/flowNodes->GetDensity(iPoint));
+              } else {
+                su2double VorticityMag = max(GeometryToolbox::Norm(3, flowNodes->GetVorticity(iPoint)), 1e-12);
+                FrictionVelocity = sqrt(flowNodes->GetLaminarViscosity(iPoint)*VorticityMag);
```

Comment:
In the original paper from Spalart and Allmaras they propose the formulation for the friction velocity specifically for a 3D problem. Since vorticity is zero for 2D problems, I assumed that I could use the standard definition for the friction velocity as done for the SST model. Probably the result is the same, but I noticed strange values for the turbulence index with SST. On the contrary, SA seems to give more reasonable results, aligned with the theory.

Reply:
Thanks, leave that comment in the code ;)